ratelimits/limit.go

```diff
@@ -0,0 +1,105 @@
+package ratelimits
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/letsencrypt/boulder/config"
+	"github.com/letsencrypt/boulder/strictyaml"
+)
+
+type limit struct {
+	// Burst specifies maximum concurrent allowed requests at any given time. It
+	// must be greater than zero.
+	Burst int64
+
+	// Count is the number of requests allowed per period. It must be greater
+	// than zero.
+	Count int64
+
+	// Period is the duration of time in which the count (of requests) is
+	// allowed. It must be greater than zero.
+	Period config.Duration
+}
+
+type limits map[string]limit
+
+func parseDefaultName(k string) (string, error) {
+	name, ok := stringToName[k]
+	if !ok {
+		return "", fmt.Errorf(
+			"unrecognized limit %q, must be one of %v", k, limitNames)
```

Comment:
Suppose we're parsing a file of overrides, and one of them is misformatted and doesn't contain a colon. Then parseLimits will jump into its "Default limit" code-path, call this function, and return this error. Is this a helpful error for someone trying to figure out why their overrides file isn't loading? I think a much more helpful error would be one of the errors from the function below, stating that an override must be formatted like "name:id". This is why I think it would be nice to have two separate "parseDefaults" and "parseOverrides" functions, rather than one shared function which doesn't know which of those two tasks it's supposed to be doing.

Reply:
Yeah this makes sense, I'll refactor.